### PR TITLE
Python testing for MSVC

### PR DESCRIPTION
**Pull request by isuruf**
Added python testing for MSVC in Release mode.

In Debug mode, `python27_d.lib` which is the debug version of python is missing.

### CODE REVIEW

File: symengine/python/symengine/tests/test_ntheory.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def test_prime_factors():
     assert prime_factors(1) == []
 
 def test_prime_factor_multiplicities():
-    assert prime_factor_multiplicities(90) == {2: 1, 3: 2, 5: 1}
+    assert prime_factor_multiplicities(90) == {Integer(2): 1, Integer(3): 2, Integer(5): 1}
```
Comment (certik) [inline symengine/python/symengine/tests/test_ntheory.py]: This is fine for now to make the tests pass, but it'd be nice to figure out a way to make the original code work in Python, because it's natural. I don't know if there is a way or not.

**Comment by certik** [comment]
That looks good to me.

File: symengine/python/symengine/tests/test_ntheory.py (hunk 1 above)
Comment (isuruf) [inline symengine/python/symengine/tests/test_ntheory.py]: Fixed this in the commit

**Comment by certik** [comment]
I think the last change is fine. It might even speed some things up.

**Comment by isuruf** [comment]
@certik, I added in a change to make the hashes of SymEngine::Integer and python int the same.

**Comment by certik** [comment]
Cool! However, isn't the Python hash implementation dependent?

**Comment by isuruf** [comment]
Not sure. That's why I pushed in this change to see if the new tests fail.

**Comment by isuruf** [comment]
This is ready for review

File: symengine/integer.cpp [hunk 2]
```diff
@@ -10,10 +10,9 @@ Integer::Integer(int i)
 
 std::size_t Integer::__hash__() const
 {
-    std::hash<long long int> hash_fn;
     // only the least significant bits that fit into "signed long int" are
     // hashed:
-    return hash_fn(this->i.get_si());
+    return this->i.get_ui() * mpz_sgn(this->i.get_mpz_t());
```
Comment (certik) [inline symengine/integer.cpp]: So why is this change still needed? You showed that the hashing is actually implementation dependent, and so we can't make our hashes equal to the Python hashes.

So it seems the motivation for this change might be speed, or perhaps it's not even faster? Either way, if it is not related to this PR, I would not do this as par of this PR.

Comment (isuruf) [inline symengine/integer.cpp]: For ints with small absolute values hashes are the same in all python versions (https://docs.python.org/3/library/stdtypes.html#hashing-of-numeric-types), so that tests like `set((2*x**2*y).args) == set((2, x**2, y))` can be kept as it is.

I'll look into whether speed is a concern. I'll revert back to 08ecc0b123548e9e39f74f39383b8c89b78e4352, get this merged in and then see what should be done.

Comment (certik) [inline symengine/integer.cpp]: Perfect.

Comment (isuruf) [inline symengine/integer.cpp]: Done

**Comment by certik** [comment]
+1 to merge if tests pass.
